test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,840 @@ TEST_F(SearchFilter, ExcludeRamp) {
+         std::to_string(closure_map.nodes.at("A").lat()) %
+         std::to_string(closure_map.nodes.at("A").lng()) %
+         std::to_string(closure_map.nodes.at("E").lat()) %
+         std::to_string(closure_map.nodes.at("E").lng()) % costing % costing % date_type)
+            .str();
+
+    EXPECT_THROW(
+        {
+          try {
+            gurka::route(closure_map, bad_request, reader);
+          } catch (const valhalla_exception_t& e) {
+            EXPECT_EQ(e.code, 143);
+            EXPECT_STREQ(
+                "ignore_closure in costing and exclude_closure in search_filter cannot both be specified",
+                e.what());
+            throw;
+          }
+        },
+        valhalla_exception_t);
+  }
+  // ignore_closures:true & exclude_closures:false on one location
+  {
+    const std::string& bad_request =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s},{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"], "ignore_closures":false}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("A").lat()) %
+         std::to_string(closure_map.nodes.at("A").lng()) %
+         std::to_string(closure_map.nodes.at("C").lat()) %
+         std::to_string(closure_map.nodes.at("C").lng()) %
+         std::to_string(closure_map.nodes.at("E").lat()) %
+         std::to_string(closure_map.nodes.at("E").lng()) % costing % costing % date_type)
+            .str();
+
+    EXPECT_THROW(
+        {
+          try {
+            gurka::route(closure_map, bad_request, reader);
+          } catch (const valhalla_exception_t& e) {
+            EXPECT_EQ(e.code, 143);
+            EXPECT_STREQ(
+                "ignore_closure in costing and exclude_closure in search_filter cannot both be specified",
+                e.what());
+            throw;
+          }
+        },
+        valhalla_exception_t);
+  }
+}
+
+// Generate different combinations of costing (auto, bus, etc) & date_type (0, 3)
+std::vector<costing_and_datetype> buildParams() {
+  std::vector<costing_and_datetype> params;
+
+  std::vector<std::string> costings = {"auto",  "motorcycle", "motor_scooter", "bus",
+                                       "truck", "hov",        "taxi"};
+  for (const auto& costing : costings) {
+    // Add date_type:3 for time-invariant bidir a*
+    params.emplace_back(std::make_tuple(costing, "3"));
+    // Add date_type:0 for timedep-fwd a* using current time
+    // TODO: Currently, ignoring closures at destinations does not work for
```

Comment:
Opened https://github.com/valhalla/valhalla/issues/2733 for handling closures in unidirectional astar variants.